### DISCUSSION TITLE
Why `Pending` is an array?

### DISCUSSION THREAD

**Discussion by zhaoyao91** [solidjs/solid#448] [Q&A] [answered]
see [code](https://github.com/solidui/solid/blob/v1.0.0-rc.0/packages/solid/src/reactive/signal.ts#L23)

why `Pending` variable is an array? After searching the codebase, I didn't see the usecase of as array.

**Comment by ryansolid** [comment, marked as answer]
https://github.com/solidui/solid/blob/v1.0.0-rc.0/packages/solid/src/reactive/signal.ts#L667
https://github.com/solidui/solid/blob/v1.0.0-rc.0/packages/solid/src/reactive/signal.ts#L411

More than one signal can be set in a batch. I push them to an array. Then iterate over them.

**Reply by zhaoyao91** [reply]
Sorry I missed the `q` logic. Thanks for clarifying.
